tilings/strategies/obstruction_inferral.py

```diff
@@ -34,6 +36,37 @@ def formal_step(self) -> str:
             ", ".join(str(p) for p in self.gps),
         )
 
+    def extra_parameters(
+        self, comb_class: Tiling, children: Optional[Tuple[Tiling, ...]] = None,
+    ) -> Tuple[Dict[str, str], ...]:
+        if not comb_class.extra_parameters:
+            return super().extra_parameters(comb_class, children)
+        if children is None:
+            children = self.decomposition_function(comb_class)
+            if children is None:
+                raise StrategyDoesNotApply("Strategy does not apply")
+        av = children[0]
+        av_mapped_gps = [
+            tuple(
+                av.forward_map(gp)
+                for gp in ass.gps
+                if gp.avoids(*self.gps)
+                and all(cell in av.forward_cell_map for cell in gp.pos)
+            )
+            for ass in comb_class.assumptions
+        ]
+        av_mapped_ass = [
+            TrackingAssumption(gp for gp in gps if gp.avoids(*av.obstructions))
+            for gps in av_mapped_gps
+        ]
```

Comment:
If I understand correctly this bit of code is mapping the assumption according to some cell map and then performing some cleaning based. As it seems this bit of is repeated in many the `extra_parameters` methods in should probably be factor out somewhere. Perhaps on the tracking assumption class?

Reply:
Actually, reading further I see that the entire `extra_parameters` methods is very similar on all the disjoint union based strategy. Maybe that's a track to refactoring it? I'd be happy to chat about that if you think it's a good idea.

Reply:
I changed to use the avoiding method in TrackingAssumption.

This doesn't quite belong on the level of DisjointUnionStrategy, as all maps are specific to tilings. I am open to suggestions though. 
The process is essentially three steps:
1) map according to the bijection implied by the strategy, e.g. cell insertion is identity map, point placement is the stretching map, etc
2) map this according to the forward map of the tiling initialiser
3) remove those which contain an obstruction on the child tiling

Reply:
You could argue that if css is always working with combinatorial sets, then those sets always contain combinatorial objects, which can be mapped around. However, I think css can work with more general "things", where there are no underlying objects.

Reply:
Its mostly because of 2 that it shouldn't belong there. Our strategies are really always applying two strategies at one. The strategy itself, then remove empty rows and columns.

Reply:
Ok I still think some of it should be factored as there is a lot of redundant code but I think it's part a broader design discussion we should have about how we implement maps. I'll open an issue

Reply:
Ok, I fixed conflicts. Shall we delay refactoring to a future PR then, and accept this one?

Reply:
Yes I think this is ready to merge